labs/cryptol-overview/overview.md

```diff
@@ -227,25 +192,59 @@ Q.E.D.
 (Total Elapsed Time: 0.010s, using Z3)
 ```
 
-Cryptol reports `Q.E.D.` indicating that our property is indeed true for all keys and all 16-character inputs. Cryptol currently only supports proofs of [total](https://en.wikipedia.org/wiki/Partial_function#Function) [monomorphic](https://en.wikipedia.org/wiki/Polymorphism_(computer_science)) properties with a finite domain. Here we must specify the length of the messages that we want to check this property for. This example checks the property for 16 character messages, but we could check this for at any (reasonable) length.
+Cryptol reports `Q.E.D.`, indicating that our property is indeed true for all keys and all 16-character inputs. Cryptol currently only supports proofs of [total](https://en.wikipedia.org/wiki/Partial_function#Function) [monomorphic](https://en.wikipedia.org/wiki/Polymorphism_(computer_science)) properties with a finite domain. Here we must specify the length of the messages that we want to check this property for. This example checks the property for 16 character messages, but we could check this for at any (reasonable) length.
 
 
-# Language Features
+# First Steps: Hello, Cryptol!
 
-So what makes Cryptol special compared to other langauges?
+A grand tradition when learning a new programming language is to test the waters with a Hello World program. In `helloworld.cry`, we define a function `sayHello` which is specified as follows:
 
-Cryptol is a language designed with Cryptography specifically in mind -- much of the syntax and language was designed with the way that real cryptographers think about and design systems. This allows the Cryptol user to create formal algorithm specifications that closely imitate the style used to describe these algorithms mathematically.
+```haskell
+sayHello : {a} (fin a) => [a][8] -> [7+a][8]
+sayHello name = greeting
+  where
+    greeting = "Hello, " # name
+```
 
-Furthermore, Cryptol provides direct access and easily integrates with powerful tools such as SMT solvers and the Software Analysis Workbench (SAW). These tools allow the user to *prove* facts and demonstrate properties about their code which can provide assurance guarantees that go far beyond simple unit testing.
+If you have Cryptol installed, you should be able to do the following from the terminal:
 
-We will introduce some of these features below and discuss how they support building Cryptographic specifications and evaluations. If you have access to the Cryptol interpreter you can follow along with some of the examples, but a detailed introduction to the Cryptol interpreter will be provided in future lessons.
+```shell
+$ cryptol
```

Comment:
I suggest we make this lab literate and remove the sayHello.cry file. I'm happy to take a stab at it.